app/workers/deletes.php

```diff
@@ -124,6 +127,47 @@ public function shutdown(): void
     {
     }
 
+    /**
+     * @throws Exception
+     */
+    protected function deleteSchedules(string $datetime): void
+    {
+
+        $sum = $limit = 50;
+        $offset = 0;
+        /** @var Document[] $functions */
+        while ($sum >= $limit) {
+            $schedules = $this->getConsoleDB()->find('schedules', [
+                Query::equal('region', [App::getEnv('_APP_REGION', 'default')]),
+                Query::equal('resourceType', ['function']),
+                Query::lessThanEqual('resourceUpdatedAt', $datetime),
+                Query::equal('active', [false]),
+                Query::limit($limit),
+                Query::offset($offset),
+            ]);
+
+            $sum = \count($schedules);
+            $offset = $offset + $limit;
+
+            foreach ($schedules as $schedule) {
+                    Console::info('Querying schedule for function ' . $schedule->getAttribute('resourceId'));
+                    $project = $this->getConsoleDB()->getDocument('projects', $schedule->getAttribute('projectId'));
+
+                if ($project->isEmpty()) {
+                    Console::success('Deleting schedule for function ' . $schedule->getAttribute('resourceId'));
+                    continue;
+                }
+
+                $function = $this->getProjectDB($project)->getDocument('functions', $schedule->getAttribute('resourceId'));
+
+                if ($function->isEmpty()) {
+                    $this->getConsoleDB()->deleteDocument('schedules', $schedule->getId());
+                    Console::success('Deleting schedule for function ' . $schedule->getAttribute('resourceId'));
+                }
+            }
+        }
```

Comment:
There is already a function called `deleteByGroup` which we can reuse. We don't need to do write the pagination logic again